it/controllers/CreateBusinessDetailsControllerISpec.scala

```diff
@@ -17,53 +17,108 @@
-import assets.CreateBusinessDetailsIntegrationTestConstants.{successResponse, testBusinessDetails, testIncomeSourceId}
+import assets.CreateBusinessDetailsIntegrationTestConstants._
 import helpers.ComponentSpecBase
 import helpers.servicemocks.DesCreateBusinessDetailsStub
+import models.incomeSourceDetails.CreateBusinessDetailsResponseModel.CreateBusinessDetailsErrorResponse
+import play.api.http.Status
 import play.api.http.Status._
 import play.api.libs.json.Json
 
 class CreateBusinessDetailsControllerISpec extends ComponentSpecBase {
 
 
-  "Calling the CreateBusinessDetailsController.createBusinessDetails method" when {
-    "authorised with a valid request" when {
+  "Calling CreateBusinessDetailsController.createBusinessDetails method" when {
+    "authorised with a CreateBusinessIncomeSourceRequest model" when {
       "A successful response is returned from DES" should {
-        "return a valid createBusinessDetails model" in {
+        s"return $OK response with an incomeSourceId" in {
 
           isAuthorised(true)
 
-          And("I wiremock stub a successful createBusinessDetails response")
-          DesCreateBusinessDetailsStub.stubPostDesBusinessDetails(testMtdbsa, OK, testBusinessDetails.toString, successResponse.toString)
+          DesCreateBusinessDetailsStub
+            .stubPostDesBusinessDetails(testMtdbsa, Status.OK, testCreateBusinessIncomeSourceRequest, testCreateBusinessDetailsSuccessResponse)
 
           When(s"I call POST /income-tax/income-sources/mtdbsa/$testMtdbsa/ITSA/business")
-          val res = IncomeTaxViewChange.createBusinessDetails(testMtdbsa, testBusinessDetails)
+          val res = IncomeTaxViewChange.createBusinessDetails(testMtdbsa, testCreateBusinessIncomeSourceRequest)
+
+          DesCreateBusinessDetailsStub.verifyCreateDesBusinessDetails(testMtdbsa, testCreateBusinessIncomeSourceRequest)
+
+          res should have(httpStatus(OK))
+          res.body should include(testIncomeSourceId)
+        }
+      }
+    }
+    "authorised with a CreateUKPropertyIncomeSourceRequest model" when {
+      "A successful response is returned from DES" should {
+        s"return $OK response with an incomeSourceId" in {
+
+          isAuthorised(true)
+
+          DesCreateBusinessDetailsStub
+            .stubPostDesBusinessDetails(testMtdbsa, Status.OK, testCreateUKPropertyRequest, testCreateBusinessDetailsSuccessResponse)
+
+          When(s"I call POST /income-tax/income-sources/mtdbsa/$testMtdbsa/ITSA/business")
+          val res = IncomeTaxViewChange.createBusinessDetails(testMtdbsa, testCreateUKPropertyRequest)
+
+          DesCreateBusinessDetailsStub.verifyCreateDesBusinessDetails(testMtdbsa, testCreateUKPropertyRequest)
+
+          res should have(httpStatus(OK))
+          res.body should include(testIncomeSourceId)
+        }
+      }
+    }
+    "authorised with a CreateForeignPropertyIncomeSourceRequest model" when {
+      "A successful response is returned from DES" should {
```

Comment:
Please remove all reference to DES as this is not required and we eventually we would move to IF